packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp

```diff
@@ -149,12 +139,33 @@ YogaLayoutableShadowNode::YogaLayoutableShadowNode(
         static_cast<const YogaLayoutableShadowNode&>(sourceShadowNode)
             .yogaTreeHasBeenConfigured_;
   }
+                        
+  bool wasDisplayContents = hasDisplayContentsStyle();
 
   if (fragment.props) {
     updateYogaProps();
   }
 
-  if (fragment.children) {
+  bool isDisplayContents = hasDisplayContentsStyle();
+                        
+  if (!getTraits().check(ShadowNodeTraits::Trait::LeafYogaNode) && !isDisplayContents) {
+    for (auto& child : getChildren()) {
+      if (auto layoutableChild = std::dynamic_pointer_cast<const YogaLayoutableShadowNode>(child)) {
+        runForEveryConcreteSubtree(layoutableChild, [&](const YogaLayoutableShadowNode::Shared subtreeRoot) {
+          yogaLayoutableChildren_.push_back({
+            .directChild = layoutableChild,
+            .yogaChild = subtreeRoot,
+          });
+        });
+      }
+    }
+  }
+  
+  if (isDisplayContents) {
+    yogaNode_.setChildren({});
+  }
```

Comment:
Could we fold this logic into `updateYogaChildren`? Perhaps passing previous shadow node, if we need to know before/after?

Reply:
We can but we would need to call it every time instead of when a fragment contains new children. Without either the current logic or updating `updateYogaChildren` the list would be empty for clones of the node.

After looking at it closer though I realized that I could remove iterating over subrees from there as it always happens before the method is called (https://github.com/facebook/react-native/pull/46584/commits/c14b801f8532ab62c82aee77249b23d6658eafa8)